eeps/eep-0058.md

```diff
@@ -0,0 +1,255 @@
+    Author: Sergey Prokhorov <seriy(dot)pr(at)gmail(dot)com>
+    Status: Draft
+    Type: Standards Track
+    Created: 14-Sep-2021
+    Erlang-Version: 25
+    Post-History:
+****
+EEP 58: Map comprehensions
+----
+
+
+Abstract
+========
+
+This EEP proposes syntax and semantics for map comprehensions and
+generators, similar to list and binary comprehensions and generators,
+but operating on maps and map iterators.
+
+Copyright
+=========
+
+This document has been placed in the public domain.
+
+
+Specification
+=============
+
+Maps and initial map comprehension syntax and semantics were proposed
+in EEP-43. List and binary comprehensions were specified post-fact
+in EEP-12. This EEP is based on those two EEPs.
+
+Syntax construct commonly referred as "comprehension" is in fact
+consists of 3 semi-independent parts:
+
+- "generator" - is the expression that iterates over the container,
+  matches each element of the container over pattern and binding
+  variables which can be used in "comprehensions" and "filters".
+- "comprehension" - is the expression that is executed for each
+  element produced by generator that successfully matched the pattern
+  and passed the "filter" check. Value produced by this expression
+  is appended to the resulting generated container.
+- "filter" - is the expression that is evaluated for each element
+  produced by generator that matched the pattern and returns a
+  boolean result: when it returns `false` the current element is
+  skipped from being passed to "comprehension"; when it results
+  in `true`, the element is passed to "comprehension".
+
+So, in this document we are going to keep this separation, but
+sometimes calling the whole combination of comprehension, filter and
+generator as just "comprehension", because generator syntax is not
+used outside of the comprehension context.
```

Comment:
```suggestion
So, in this document we are going to keep this separation, but
sometimes will refer to the whole combination of the comprehension, the filter and
the generator as a "comprehension", because the generator syntax is not
used outside of the comprehension context.
```